packages/lib/date-ranges.test.ts

```diff
@@ -179,6 +179,38 @@ describe("processWorkingHours", () => {
     expect(allDSTStartAt12).toBeTruthy();
     expect(allNotDSTStartAt13).toBeTruthy();
   });
+
+  it("should skip event if it ends before it starts (different days)", () => {
+    const item = {
+      days: [1, 2, 3],
+      startTime: new Date(new Date().setUTCHours(8, 0, 0, 0)), // 8 AM
+      endTime: new Date(new Date().setUTCHours(7, 0, 0, 0)), // 7 AM
+    };
+
+    const timeZone = "America/New_York";
+    const dateFrom = dayjs("2023-11-07T00:00:00Z").tz(timeZone); // 2023-11-07T00:00:00 (America/New_York)
+    const dateTo = dayjs("2023-11-08T00:00:00Z").tz(timeZone); // 2023-11-08T00:00:00 (America/New_York)
+
+    const results = processWorkingHours({ item, timeZone, dateFrom, dateTo });
+
+    expect(results).toEqual([]);
+  });
+
+  it("should skip event if it ends before it starts (same day but different hours)", () => {
+    const item = {
+      days: [1],
+      startTime: new Date(new Date().setUTCHours(8, 0, 0, 0)), // 8 AM
```

Comment:
@gitstart-calcom Thanks for adding these tests. Do we know how this happens? I'm just wondering how an end time would get set to be before a start time?

Reply:
But now, with the specific tests and the new logic, we fixed the issue and we can identify a possible issue faster 😄

Reply:
Cool yeah that's what I suspected myself. Seems like this function is being passed bad data for some reason.